### PR TITLE
ch4: code refactoring

### PR DESCRIPTION
**Pull request by yfguo**
This is the CH4 part for pmodels/mpich#3220

The main goal of the refactoring is to remove the inlining for
non-performance-critical functions. For clarification, the
``non-performance-critical'' functions refer to the functions that are
usually not used in mixed with the computation iterations in a MPI
application such as MPI_Init, comm creation, win creation, spawn, etc.
The implementation of these uninlined functions are moved to their
corresponding .c files. Specifically, this refactoring contains the
following changes and cleanups.

1. Uninlined all CH4 level functions that is not performance-critical
   and moved the implementation to .c files.
2. Remove duplicated declarications for uninlined MPID_* function from
   the .h since they are already in the `mpidch4.h`.
3. Remove namespace prefix for internal functions (the ones that only
   used in .c files). Add declarications for them to suppress warnings.
4. Remove unused functions.
5. Formatted the code as a results of name and attribute changes.

### CODE REVIEW

**Comment by hajimefu** [comment]
Could you add me in `Co-authored-by:`? 😆

**Comment by gcongiu** [comment]
test:jenkins/ch4/ofi

**Comment by gcongiu** [comment]
test:jenkins/ch4/ofi

**Comment by gcongiu** [comment]
test:jenkins/ch4/ofi

File: src/mpid/ch4/include/mpidch4.h [hunk 1]
```diff
@@ -232,7 +238,8 @@ MPIDI_CH4I_API(int, Iscatter, const void *, int, MPI_Datatype, void *, int, MPI_
 MPIDI_CH4I_API(int, Iscatterv, const void *, const int *, const int *, MPI_Datatype, void *, int,
                MPI_Datatype, int, MPIR_Comm *, MPIR_Request **);
 
-int MPID_Abort(struct MPIR_Comm *comm, int mpi_errno, int exit_code, const char *error_msg);
+MPIDI_CH4I_API_NOINLINE(int, Abort, struct MPIR_Comm *comm, int mpi_errno, int exit_code,
```
Comment (raffenet) [inline src/mpid/ch4/include/mpidch4.h]: Is this macro really necessary? I kind of think less complexity here would be a win for readability...

Comment (wesbland) [reply src/mpid/ch4/include/mpidch4.h]: Oh God, not more @charlesarcher macros... 🤢

Comment (raffenet) [reply src/mpid/ch4/include/mpidch4.h]: Right. We haven't had to touch these in quite some time. While we're here, just remove them and be done with it once and for all.

Comment (yfguo) [reply src/mpid/ch4/include/mpidch4.h]: I added the `MPIDI_CH4I_API_NOINLINE` macro just for consistency.

**Comment by gcongiu** [comment]
test:jenkins/ch4/ofi

**Comment by gcongiu** [comment]
@raffenet all the names should be updated now

**Comment by gcongiu** [comment]
test:jenkins/ch4/ofi

**Comment by yfguo** [comment]
This PR has been cherry-picked to pmodels/mpich#3220. I am closing this one and divert all further discussion to #3220.

**Pull request closed without merging**